lib/ts/Layout.cc

```diff
@@ -41,56 +47,61 @@ Layout::get()
 }
 
 void
-Layout::create(const char *prefix)
+Layout::create(ts::string_view prefix)
 {
   if (layout == nullptr) {
     layout = new Layout(prefix);
   }
 }
 
-static char *
-layout_relative(const char *root, const char *file)
+static bool
+_relative(char *path, ts::string_view root, ts::string_view file)
```

Comment:
This function exists to deal with the existing char* api as we migrate off the raw char* api.. since we are hiding this function, we should probably pass the size as @jpeach suggests. This function should also be protected: or private: